crates/rome_analyze/src/signals.rs

```diff
@@ -105,22 +119,111 @@ where
     }
 }
 
-impl<L> From<AnalyzerAction<L>> for CodeSuggestion
-where
-    L: Language,
-{
+impl<L: Language> From<AnalyzerAction<L>> for CodeSuggestionItem {
     fn from(action: AnalyzerAction<L>) -> Self {
         let (range, suggestion) = action.mutation.as_text_edits().unwrap_or_default();
 
-        CodeSuggestion {
-            span: FileSpan {
-                file: action.file_id,
-                range,
+        CodeSuggestionItem {
+            rule_name: action.rule_name,
+            category: action.category,
+            suggestion: CodeSuggestion {
+                span: FileSpan {
+                    file: action.file_id,
+                    range,
+                },
+                applicability: action.applicability,
+                msg: action.message,
+                suggestion,
+                labels: vec![],
             },
-            applicability: action.applicability,
-            msg: action.message,
-            suggestion,
-            labels: vec![],
+        }
+    }
+}
+
+impl<L: Language> AnalyzerActionIter<L> {
+    pub fn new(actions: Vec<AnalyzerAction<L>>) -> Self {
```

Comment:
Nit: Can we change this to accept any type that implements IntoIter
```suggestion
    pub fn new<I>(actions: I) -> Self where I: IntoIter<Item = AnalyzerAction<L>> {
```